### PR TITLE
246 Fix full deployment scripts

### PR DESCRIPTION
**Pull request by dospore**
### Motivation
- Front end needs to repair comms to v2 contracts
- Need a deploy script to deploy the contracts locally

### Changes
- Re-added all the contracts to FullDeploy
- Created a new folder for deployment of mock contracts in testing

### CODE REVIEW

**Comment by dospore** [comment]
Dependencies
- fix chicken and egg issue between Liquidation and TracerPerpSwaps
- remove Lib from the library names

File: deploy/FullDeploy.js [hunk 1]
```diff
@@ -4,17 +4,67 @@ module.exports = async function (hre) {
...
-    await deploy("LibLiquidationMock", {
+    // deploy oracles
+    await deploy('Oracle', {
         from: deployer,
         log: true,
+    });
+
+    // deploy token
+    await deploy('TestToken', {
+        args: ["100000000000000000000000"],
```
Comment (sporejack) [inline deploy/FullDeploy.js:45]: Magic number

File: scripts/DeployAndAddTracer.js [hunk 2]
```diff
@@ -1,67 +1,54 @@
...
+    //         liquidation.address,
+    //         maxLeverage,
+    //         fundingRateSensitivity,
+    //         feeRate
+    //     ]
+    // )
+
+    // await factory.deployTracer(deployTracerData)
+    // let tracerAddr = await factory.tracersByIndex(0)
+    console.log(`Factory Deplpoyed: ${factory.address}`)
```
Comment (sporejack) [inline scripts/DeployAndAddTracer.js]:
```suggestion
    console.log(`Factory Deployed: ${factory.address}`)
```

File: scripts/DeployAndAddTracer.js [hunk 3]
```diff
@@ -1,67 +1,54 @@
...
+    // console.log(`Tracer Deplpoyed: ${tracerAddr}`)
+    console.log(`Trader Deplpoyed: ${trader.address}`)
```
Comment (sporejack) [inline scripts/DeployAndAddTracer.js]:
```suggestion
    console.log(`Trader Deployed: ${trader.address}`)
```

File: scripts/DeployAndAddTracer.js [hunk 4]
```diff
@@ -1,67 +1,54 @@
...
     console.log(`Margin Token Deplpoyed: ${token.address}`)
```
Comment (sporejack) [inline scripts/DeployAndAddTracer.js]:
```suggestion
    console.log(`Margin Token Deployed: ${token.address}`)
```

File: scripts/DeployAndAddTracer.js [hunk 5]
```diff
@@ -1,67 +1,54 @@
...
-            pricing.address,
-            maxLeverage,
-            1, //funding rate sensitivity
-        ]
-    )
-
-    await factory.deployTracer(deployTracerData)
-    let tracerAddr = await factory.tracersByIndex(0)
-    console.log(`Tracer Deplpoyed: ${tracerAddr}`)
+    // var deployTracerData = ethers.utils.defaultAbiCoder.encode(
```
Comment (sporejack) [inline scripts/DeployAndAddTracer.js]: If these are commented out, should we take this opportunity to simplify our deployment code now too? @raymogg

File: scripts/DeployAndAddTracer.js [hunk 6]
```diff
@@ -1,67 +1,54 @@
-const hre = require("hardhat")
-const TestToken = artifacts.require("TestToken")
-const { BN, time } = require("@openzeppelin/test-helpers")
-const Gov = artifacts.require("Gov")
-const TracerFactory = artifacts.require("TracerPerpetualsFactory")
+const hre = require("hardhat");
+const { BN } = require("@openzeppelin/test-helpers");
```
Comment (sporejack) [inline scripts/DeployAndAddTracer.js]: Does `ethers` not provide its own `BigNumber` implementation?

**Review by sporejack — CHANGES_REQUESTED** [review]
Several things needing attention; most minor.

**Comment by sporejack** [comment]
@dospore 
```sh
$ yarn run lint:fix
$ git add .
$ git commit
$ git push
```

File: scripts/DeployAndAddTracer.js (hunk 6 above)
Comment (dospore) [reply scripts/DeployAndAddTracer.js]: this was just legacy, ethers does ill use that instead

**Review by CalabashSquash — COMMENTED** [review]
Looks good once @sporejack 's comments are addressed

**Pull request merged**